Users deletion (backend)

EvanAntoni opened the pull request:
User delete feature. Pull request into branch users_list.

EvanAntoni committed:
Add user deletion

MaxSukhovetskyi commented on easyrest/models/user.py:
```diff
@@ -70,3 +70,12 @@ def add(database, form_data):
+        :param database: Database session.
+        :param user: Instance of deletable user.
+        """
+        database.delete(user)
```
In this particular case you can do delete method like this:

def delete(self, session):
      session.delete(self)

So no need to write @staticmethod

EvanAntoni replied:
It makes sense. Thank's.
But what's wrong with the`@staticmethod`?

MaxSukhovetskyi replied:
Its confusing)

EvanAntoni replied:
Guess which function deletes the user?
Function 1:
```
def delete(self, session):
    session.delete(self)
```
Function 2:
```
def delete(database, user):
    database.delete(user)
```

serge30 replied:
Both of them are confusing. I do not see any reasons for such functions, as they do not use any context. You may just inline its body in place of use.

Using in controller something like is acceptable and clear:
```
request.dbsession.delete(user)
```

EvanAntoni replied:
I made this function with some expectation for the future. For example, if you need to delete files associated with a user (well, avatar or something else), you can simply do something like this to the delete method:
```
def delete(database, user):
    storage.delete(user.avatar)
    database.delete(user)
```

EvanAntoni committed:
Edit user deletion

Simplify user deletion logic. The deletion method is removed
from the user model and implemented in the controller.

EvanAntoni committed:
Merge branch 'users_list' into users_delete
